### PR TITLE
Write Tests for Visa DAO

### PR DESCRIPTION
**Pull request by snwessel**
Follow-up tests for POL-37

### CODE REVIEW

File: service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.Collections;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.transaction.annotation.Transactional;
+
+public class VisaDAOTest extends BaseTest {
```
Comment (dvoet) [inline service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java]: name should be `GA4GHVisaDAOTest`

File: service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    visa =
+        GA4GHVisa.builder()
+            .visaType("fake")
+            .expires(new Timestamp(100))
+            .jwt("fake-jwt")
+            .issuer("issuer")
+            .tokenType(TokenTypeEnum.access_token)
+            .lastValidated(new Timestamp(10))
+            .build();
+  }
```
Comment (dvoet) [inline service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java]: Why do we need to build a new one ever test iteration? Why not make `visa` final and initialize it at declaration time?

File: service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    assertTrue(savedVisa1.getId() > 0);
+    assertTrue(savedVisa2.getId() > 0);
+    assertEquals(expectedVisa1, savedVisa1.withId(0));
+    assertEquals(expectedVisa2, savedVisa2.withId(0));
+
+    var loadedVisas = visaDAO.listVisas(passportId);
+    assertEquals(loadedVisas.size(), 2);
+    assertEquals(savedVisa1, loadedVisas.get(0));
+    assertEquals(savedVisa2, loadedVisas.get(1));
```
Comment (dvoet) [inline service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java]: I don't think the test should rely on the order. I would do
```assertEquals(Set.of(savedVisa1, savedVisa2), Set.copyOf(loadedVisas))```

**Review by Shakespeared — APPROVED** [review]

**Review by dvoet — CHANGES_REQUESTED** [review]

File: service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java (hunk 1 above)
Comment (snwessel) [reply service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java]: Sounds good! And I also just renamed `PassportDAOTest` to `GA4GHPassportDAOTest` for consistency.

**Review by dvoet — APPROVED** [review]

File: service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java (hunk 1 above)
Comment (dvoet) [reply service/src/test/java/bio/terra/externalcreds/dataAccess/VisaDAOTest.java]: sure

**Pull request merged**